bigquery_etl/cli/utils.py

```diff
@@ -251,3 +251,42 @@ def temp_dataset_option(
         help="Dataset where intermediate query results will be temporarily stored, "
         "formatted as PROJECT_ID.DATASET_ID",
     )
+
+
+def extract_last_group_by_from_query(query_file=None, query=None):
+    """Return the list of columns in the latest group by of a query."""
+    if query_file:
```

Comment:
This is not tested at all ... also what format do you expect the query_file parameter to be in?

Reply:
It expects a string. This will be called from the backfill command where it's validated in advance. Do you want me to add test at this point?